core/txpool/txpool.go

```diff
@@ -81,6 +89,44 @@ func New(gasTip *big.Int, chain BlockChain, subpools []SubPool) (*TxPool, error)
 	return pool, nil
 }
 
+// reserver is a method to create an address reservation callback to exclusively
+// assign/deassign addresses to/from subpools. This can ensure that at any point
+// in time, only a single subpool is able to manage an account, avoiding cross
+// subpool eviction issues and nonce conflicts.
+func (p *TxPool) reserver(subpool SubPool) AddressReserver {
+	return func(addr common.Address, reserve bool) error {
+		p.reserveLock.Lock()
+		defer p.reserveLock.Unlock()
+
+		owner, exists := p.reservations[addr]
```

Comment:
Would it make sense to have a `meter` on the size of `p.reservations`? In case we miss an unreserve somewhere, this could baloon over a few weeks. Would be neat to be able to spot it (?)

Reply:
Good idea

Reply:
yep, sounds reasonable to me

Reply:
Done.